src/main/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplication.scala

```diff
@@ -139,25 +143,135 @@ private[spark] class ArmadaClientApplication extends SparkApplication {
 
     // # FIXME: Need to check how this is launched whether to submit a job or
     // to turn into driver / cluster manager mode.
-    val jobId = submitDriverJob(armadaClient, clientArguments, sparkConf)
+    val (driverJobId, executorJobIds) = submitArmadaJob(armadaClient, clientArguments, sparkConf)
 
     val lookoutBaseURL = sparkConf.get(ARMADA_LOOKOUTURL)
     val lookoutURL =
       s"$lookoutBaseURL/?page=0&sort[id]=jobId&sort[desc]=true&" +
-        s"ps=50&sb=$jobId&active=false&refresh=true"
-    log(s"Lookout URL for this job is $lookoutURL")
+        s"ps=50&sb=$driverJobId&active=false&refresh=true"
+    log(s"Lookout URL for the driver job is $lookoutURL")
 
     ()
   }
 
-  private def getExecutorContainers(
-      numExecutors: Int,
-      driverServiceName: String,
+  private def validateConfig(conf: SparkConf): Unit = {
+    val nodeSelectors       = commaSeparatedLabelsToMap(conf.get(ARMADA_JOB_NODE_SELECTORS))
+    val gangUniformityLabel = conf.get(ARMADA_JOB_GANG_SCHEDULING_NODE_UNIFORMITY)
+
+    if (nodeSelectors.isEmpty && gangUniformityLabel.isEmpty) {
+      throw new IllegalArgumentException(
+        "Either node selectors or gang scheduling node uniformity label must be set."
+      )
+    }
+  }
+
+  private def submitArmadaJob(
+      armadaClient: ArmadaClient,
+      clientArguments: ClientArguments,
       conf: SparkConf
-  ): Seq[Container] = {
-    (0 until numExecutors).map {
-      getExecutorContainer(_, driverServiceName, conf)
+  ): (String, Seq[String]) = {
+    val queue               = conf.get(ARMADA_JOB_QUEUE)
+    val jobSetId            = conf.get(ARMADA_JOB_SET_ID)
+    val (driver, executors) = newSparkJobSubmitRequestItems(clientArguments, conf)
```

Comment:
it would be could to have unit tests that serialize these outputs and confirm that the values are as exepected both the default values and explicitly set ones.

Reply:
Can you elaborate a bit more? I don't fully understand

Reply:
I'll take care of it in my PR.